tool/prdeployer.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:firebase/firebase_io.dart';
+import 'dart:async';
+import 'package:github/server.dart';
+import 'dart:io';
+
+GitHub github;
+String firebaseAuth;
+PullRequest currentPullRequest;
+
+String firebaseRoot = "https://flutter-web-controller.firebaseio.com/flutter/pulls";
+
+main(List<String> arguments) async {
+  Map<String, String> env = Platform.environment;
+  github = createGitHubClient(auth: new Authentication.withToken(env["GITHUB_TOKEN"]));
+  
+  firebaseAuth = arguments[1];
+  String branchName = arguments[0];
+
+  FirebaseClient fbClient = new FirebaseClient(firebaseAuth);
+
+  List<PullRequest> prsList = await getPullRequests();
+  currentPullRequest = prsList.firstWhere((p) => p.head.ref == branchName);
+
+  String rootPath = "${firebaseRoot}.json";
+
+  Map<String, Map<String, String>> projectsData = await fbClient.get(rootPath);
+  String projectToDeploy;
+
+  await projectsData.forEach((String key, Map<String, String> value) async {
+    if (value["branch"] == branchName) {
+      // this has already been deployed.
+      // deploy again
+      projectToDeploy = value["name"];
+    } else {
+      bool oldBranch = await isBranchOld(value["branch"]);
+      if (oldBranch) {
+        await fbClient.patch("${firebaseRoot}/$key.json", {"branch": null});
+      }
+    }
+  });
+
+  if (projectToDeploy == null) {
+    await projectsData.forEach((String key, Map<String, String> value) async {
+      if (projectToDeploy != null) {
+        // project already set
+        // todo: remove this foreach with a better iterator pattern
+        return;
+      }
+      projectToDeploy = value["name"];
+      await fbClient.patch("${firebaseRoot}/$key.json", {"branch": branchName});
+      await postLinkToGithub(projectToDeploy, currentPullRequest);
+    });
+  }
+
+  print(projectToDeploy);
```

Comment:
The print value is used in travis.sh to populate firebase deploy, so it just needs to be on its own